src/v0.8/core/finance/Finance.sol

```diff
@@ -339,29 +402,68 @@ contract Finance is
         FinanceType.PaymentInfo[] memory paymentsInfo
     ) internal onlySupportToken(_token) {
         for (uint256 i = 0; i < paymentsInfo.length; i++) {
-            financeAccount[_datasetId][_matchingId][paymentsInfo[i].payer][
-                _token
-            ]._payment(_type, paymentsInfo[i].amount);
-
-            for (uint256 j = 0; j < paymentsInfo[i].payees.length; j++) {
-                financeAccount[_datasetId][_matchingId][
-                    paymentsInfo[i].payees[j].payee
-                ][_token]._income(_type, paymentsInfo[i].payees[j].amount);
-
-                if (
-                    paymentsInfo[i].payees[j].payee ==
-                    roles.filplus().getBurnAddress()
-                ) {
+            if (paymentsInfo[i].amount != 0) {
+                // Payer account payment.
+                financeAccount[_datasetId][_matchingId][paymentsInfo[i].payer][
+                    _token
+                ]._payment(_type, paymentsInfo[i].amount);
+
+                for (uint256 j = 0; j < paymentsInfo[i].payees.length; j++) {
+                    // Payee account income.
                     financeAccount[_datasetId][_matchingId][
-                        paymentsInfo[i].payer
-                    ][_token]._burn(paymentsInfo[i].payees[j].amount);
-
-                    SendAPI.send(
-                        FilAddresses.fromEthAddress(
-                            roles.filplus().getBurnAddress()
-                        ),
-                        paymentsInfo[i].payees[j].amount
-                    );
+                        paymentsInfo[i].payees[j].payee
+                    ][_token]._income(_type, paymentsInfo[i].payees[j].amount);
+
+                    // Payee account is burnAddress, burn process.
+                    if (
+                        paymentsInfo[i].payees[j].payee ==
+                        roles.filplus().getBurnAddress()
+                    ) {
+                        financeAccount[_datasetId][_matchingId][
+                            paymentsInfo[i].payer
+                        ][_token]._burn(paymentsInfo[i].payees[j].amount);
+
+                        SendAPI.send(
+                            FilAddresses.fromEthAddress(
+                                roles.filplus().getBurnAddress()
+                            ),
+                            paymentsInfo[i].payees[j].amount
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+    /// @dev Internal function for processing sub account escrow.
+    /// @param _datasetId The ID of the dataset.
+    /// @param _subAccountMatchingId The ID of the matching.
+    /// @param _token The address of the token used for the payment.
+    /// @param _type The FinanceType.Type specifying the type of payment.
+    /// @param paymentsInfo An array of FinanceType.PaymentInfo containing payment details.
+    function _subAccountEscrowProcess(
```

Comment:
The suggestion is to rename _subAccountEscrowProcess to _moveEscrowFundsProcess.